### PR TITLE
#33714 move BRP to new module: haalcentraal-brp

### PR DESCRIPTION
**Pull request by Klaas-Ritense**
Screenshot 2022-04-13 at 09 37 08

### CODE REVIEW

File: haalcentraal/haalcentraal-brp/src/main/kotlin/com/ritense/haalcentraal/brp/connector/HaalCentraalBrpConnector.kt [hunk 1]
```diff
@@ -14,35 +14,35 @@
...
 import kotlinx.coroutines.runBlocking
 import java.security.InvalidParameterException
 import java.time.LocalDate
 import java.time.format.DateTimeFormatter
 
-@ConnectorType(name = "HaalCentraal")
-class HaalCentraalBRPConnector(
-    private var haalCentraalBRPProperties: HaalCentraalBRPProperties,
-    private var haalCentraalBRPClient: HaalCentraalBRPClient
+@ConnectorType(name = "HaalCentraalBrp")
```
Comment (Rick-Ritense) [inline haalcentraal/haalcentraal-brp/src/main/kotlin/com/ritense/haalcentraal/brp/connector/HaalCentraalBrpConnector.kt:31]: Moet er een migratiescript komen om dit automatisch om te zetten in een bestaande database?

Comment (Rick-Ritense) [reply haalcentraal/haalcentraal-brp/src/main/kotlin/com/ritense/haalcentraal/brp/connector/HaalCentraalBrpConnector.kt:31]: oh never mind I see it already exists

Comment (Rick-Ritense) [reply haalcentraal/haalcentraal-brp/src/main/kotlin/com/ritense/haalcentraal/brp/connector/HaalCentraalBrpConnector.kt:31]: I see one it exists for the properties but not the type

Comment (Klaas-Ritense) [reply haalcentraal/haalcentraal-brp/src/main/kotlin/com/ritense/haalcentraal/brp/connector/HaalCentraalBrpConnector.kt:31]: I did add them [here](https://github.com/valtimo-platform/valtimo-backend-libraries/blob/feature/33714_API_BRP_specificeren_in_HaalCentraal_code/haalcentraal/haalcentraal-brp/src/main/resources/config/liquibase/changelog/20220413-rename-connector-to-haalcentraal-brp.xml#L23-L25)

File: docker-resources/docker-compose-base-test.yml [hunk 2]
```diff
@@ -39,6 +39,6 @@ services:
 #   url: jdbc:mysql://localhost:3316/wordpress-mail-test
 #   url: jdbc:mysql://localhost:3317/smartdocuments-test
 #   url: jdbc:mysql://localhost:3318/besluit-test
-#   url: jdbc:mysql://localhost:3319/haalcentraal-test
+#   url: jdbc:mysql://localhost:3319/haalcentraal-bpr-test
```
Comment (ivo-ritense) [inline docker-resources/docker-compose-base-test.yml:42]: brp

File: haalcentraal/haalcentraal-brp/src/main/kotlin/com/ritense/haalcentraal/brp/autoconfigure/HaalCentraalLiquibaseAutoConfiguration.kt [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+import org.springframework.core.Ordered.HIGHEST_PRECEDENCE
+import org.springframework.core.annotation.Order
+
+@Configuration
+internal class HaalCentraalLiquibaseAutoConfiguration {
+
+    @Order(HIGHEST_PRECEDENCE + 21)
+    @Bean
+    @ConditionalOnMissingBean(name = ["haalcentraalBrpLiquibaseMasterChangeLogLocation"])
+    fun haalcentraalBrpLiquibaseMasterChangeLogLocation(): LiquibaseMasterChangeLogLocation? {
```
Comment (ivo-ritense) [inline haalcentraal/haalcentraal-brp/src/main/kotlin/com/ritense/haalcentraal/brp/autoconfigure/HaalCentraalLiquibaseAutoConfiguration.kt:33]: Why is the rerturn type nullable?

File: haalcentraal/haalcentraal-brp/src/main/resources/config/liquibase/changelog/20220413-rename-connector-to-haalcentraal-brp.xml [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+    <changeSet author="Ritense" id="1">
+        <update tableName="connector_type">
+            <column name="name" value="HaalCentraalBrp" type="VARCHAR(512)"/>
+            <where>name = 'HaalCentraal'</where>
+        </update>
+    </changeSet>
+
+    <changeSet author="Ritense" id="2">
+        <update tableName="connector_type">
+            <column name="connector_properties" type="${jsonType}" valueComputed="REPLACE(connector_properties, 'com.ritense.haalcentraal.connector.HaalCentraalBRPProperties', 'com.ritense.haalcentraal.brp.connector.HaalCentraalBrpProperties')" />
```
Comment (ivo-ritense) [inline haalcentraal/haalcentraal-brp/src/main/resources/config/liquibase/changelog/20220413-rename-connector-to-haalcentraal-brp.xml:31]: Does this also work for postgres?

Comment (Klaas-Ritense) [reply haalcentraal/haalcentraal-brp/src/main/resources/config/liquibase/changelog/20220413-rename-connector-to-haalcentraal-brp.xml:31]: [It should](https://www.postgresqltutorial.com/postgresql-string-functions/postgresql-replace/)

But I've only tested it for mysql

File: haalcentraal/haalcentraal-brp/src/main/kotlin/com/ritense/haalcentraal/brp/autoconfigure/HaalCentraalLiquibaseAutoConfiguration.kt (hunk 3 above)
Comment (Klaas-Ritense) [reply haalcentraal/haalcentraal-brp/src/main/kotlin/com/ritense/haalcentraal/brp/autoconfigure/HaalCentraalLiquibaseAutoConfiguration.kt:33]: fixed

File: docker-resources/docker-compose-base-test.yml (hunk 2 above)
Comment (Klaas-Ritense) [reply docker-resources/docker-compose-base-test.yml:42]: fixed

File: haalcentraal/haalcentraal-brp/src/main/kotlin/com/ritense/haalcentraal/brp/connector/HaalCentraalBrpConnector.kt (hunk 1 above)
Comment (Rick-Ritense) [reply haalcentraal/haalcentraal-brp/src/main/kotlin/com/ritense/haalcentraal/brp/connector/HaalCentraalBrpConnector.kt:31]: ah sorry missed that one

**Pull request merged**